File: сross_platform_software_systems/14/user_service/user_service/src/main/java/com/c3equalz/user_service/application/common/ports/IdentityProvider.java
package com.c3equalz.user_service.application.common.ports;

import com.c3equalz.user_service.domain.user.values.UserID;
import reactor.core.publisher.Mono;

/**
 * Interface for getting current user identity.
 * For more information, see: <a href="https://t.me/advice17_chat/4929">Telegram Tishka17</a>
 */
public interface IdentityProvider {
    /**
     * Gets the current user ID from the security context.
     *
     * @return Mono containing the current user ID
     */
    Mono<UserID> getCurrentUserID();
}

File: сross_platform_software_systems/14/chat_service/src/chat_service/application/common/ports/identity_provider.py
from abc import abstractmethod
from typing import Protocol

from chat_service.domain.user.values.user_id import UserID


class IdentityProvider(Protocol):
    @abstractmethod
    async def get_current_user_id(self) -> UserID:
        """
        Interface for getting user info.
        For more information, see: https://t.me/advice17_chat/4929
        :raises AuthenticationError:
        """
        ...
